crates/router/src/connector/utils.rs

```diff
@@ -289,6 +289,32 @@ fn get_card_issuer(card_number: &str) -> Result<CardIssuer, Error> {
         errors::ConnectorError::NotImplemented("Card Type".into()),
     ))
 }
+pub trait WalletData {
+    fn get_wallet_token(&self) -> Result<String, Error>;
+    fn get_wallet_token_as_json<T>(&self) -> Result<T, Error>
+    where
+        T: serde::de::DeserializeOwned;
+}
+
+impl WalletData for api::WalletData {
```

Comment:
```suggestion
impl api::WalletData {
```

and remove the trait WalletData

Reply:
cannot define inherent `impl` for a type outside of the crate where the type is defined
define and implement a trait or new type instead